src/core/recv_buffer.c

```diff
@@ -8,10 +8,8 @@
     The receive buffer is a dynamically sized circular buffer for reassembling
     stream data and holding it until it's delivered to the client.
 
-    When the buffer is resized, all bytes in the buffer are copied to a new
```

Comment:
Has this comment been out-of-sync for a while now?
This change seems to suggest that this copy-and-realloc behavior has been deprecated for some time.
If not, consider keeping some form of it.